File: discord-music-bot/src/main/java/net/masterzach32/swagbot/api/jokes/YMJoke.java
package net.masterzach32.swagbot.api.jokes;

import com.mashape.unirest.http.HttpResponse;
import com.mashape.unirest.http.JsonNode;
import com.mashape.unirest.http.Unirest;
import com.mashape.unirest.http.exceptions.UnirestException;

public class YMJoke implements IRandomJoke {

	private String joke;

	public YMJoke() {
		try {
			HttpResponse<JsonNode> json = Unirest.get("http://api.yomomma.info/").asJson();
			joke = json.getBody().getArray().getJSONObject(0).getString("joke");
		} catch (UnirestException e) {
			e.printStackTrace();
		}
	}
	
	public String getJoke() {
		return joke;
	}
}

File: discord-music-bot/src/main/java/net/masterzach32/swagbot/api/jokes/YMJoke.kt
package net.masterzach32.swagbot.api.jokes

import com.mashape.unirest.http.Unirest
import com.mashape.unirest.http.exceptions.UnirestException

class YMJoke: IRandomJoke {

    private var joke: String = ""

    init {
        try {
            val json = Unirest.get("http://api.yomomma.info/").asJson()
            joke = json.body.array.getJSONObject(0).getString("joke")
        } catch (e: UnirestException) {
            e.printStackTrace()
        }

    }

    override fun getJoke(): String {
        return joke
    }
}